addCohortIntersect and others

KimLopezGuell opened the pull request.

catalamarti commented:
Also as agreed with @edward-burn could you change the name to addIntersect please?

KimLopezGuell commented:
I have corrected many of the remarks pointed out. I have changed quite some things in the function, allowing it to be more general and used in both cohorts and tables.

There are still some issues unresolved:
- I only compute the permanent table in the end, if tablePrefix = TRUE. Otherwise throughout the code I work with temporary tables
- I have not looked yet at making this quick
- There is one test that does not pass: when using a table as the base cohort, the indexDate is not "cohort_start_date", and the function breaks. I do not know why and have not been able to correct it yet. You can see my example which fails using "condition_occurrence" as base and "drug_exposure" as target. Line 536 of the test-addIntersect.R.

I will work on it more next week, but wanted to upload the failing script if you wanted to take a look.

catalamarti commented:
Hi @KimLopezGuell  it would be nice if you take a look. I merge so we can continue the development of the package
